core/src/main/java/org/apache/calcite/sql/type/SqlTypeName.java

```diff
@@ -309,7 +314,10 @@ public enum SqlTypeName {
    * matches the given name, or throws {@link IllegalArgumentException}; never
    * returns null. */
   public static SqlTypeName lookup(String tag) {
-    String tag2 = tag.replace(' ', '_');
+    // Special handling for TIME WITH TIME ZONE and
+    // TIMESTAMP WITH TIME ZONE, whose names are TIME_TZ and TIMESTAMP_TZ
+    final String tag1 = tag.replace("WITH TIME ZONE", "TZ");
```

Comment:
This is kind of personal preference but you could simplify this with a single regex like 
`  return re.sub(r"(?i)WITH TIME ZONE|\s", "TZ", tag)`

Reply:
Since this is not performance-critical I think I prefer the version I wrote, because it is obvious what it does. To understand your version you have to be an expert in regular expressions, and most people aren't. Your version seems to be case-insensitive, but I think that this string will always be uppercase, since it is inserted here by the parser. I will document this behavior.

Reply:
+1 mihai's perspective